src/main/java/teammates/ui/controller/AdminSearchPageAction.java

```diff
@@ -87,22 +87,22 @@ private AdminSearchPageData putCourseNameIntoMap(List<StudentAttributes> student
                                                      List<InstructorAttributes> instructors,
                                                      AdminSearchPageData data) {
         for (StudentAttributes student : students) {
-            if (student.course != null && !data.courseIdToCourseNameMap.containsKey(student.course)) {
+            if (student.course != null) {
                 CourseAttributes course = logic.getCourse(student.course);
                 if (course != null) {
                     //TODO: [CourseAttribute] remove desanitization after data migration
-                    data.courseIdToCourseNameMap.put(
+                    data.courseIdToCourseNameMap.putIfAbsent(
```

Comment:
To me it seems that this change will incur a potential slowdown (by a little bit i guess) due to the previously avoidable `logic.getCourse(student.course)`, but the logic is clearer :)